File: generated-code/scala/example/FileReadWrite/src/main/scala/model/Enumeration.scala
package model

import util.StreamUtil

/**
 * Example enumeration
 */
sealed abstract class Enumeration (val tag: Int) {
    /**
     * Write Enumeration to output stream
     */
    def writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, tag)
    }
}

object Enumeration {
    /**
     * First option
     */
    case object VALUE_ONE extends Enumeration(0)
    /**
     * Second option
     */
    case object VALUE_TWO extends Enumeration(1)

    /**
     * Read Enumeration from input stream
     */
    def readFrom(stream: java.io.InputStream): Enumeration =
        StreamUtil.readInt(stream) match {
            case VALUE_ONE.tag => VALUE_ONE
            case VALUE_TWO.tag => VALUE_TWO
            case _ => throw new java.io.IOException("Unexpected tag value")
        }
}

File: generated-code/javascript/example/FileReadWrite/model/enumeration.js
/**
 * Example enumeration
 */
class Enumeration {
    constructor(name, tag) {
        this.name = name;
        this.tag = tag;
    }

    /**
     * First option
     */
    static VALUE_ONE = new Enumeration("VALUE_ONE", 0);
    /**
     * Second option
     */
    static VALUE_TWO = new Enumeration("VALUE_TWO", 1);

    /**
     * Read Enumeration from input stream
     */
    static async readFrom(stream) {
        const tag = await stream.readInt();
        if (tag == Enumeration.VALUE_ONE.tag) {
            return Enumeration.VALUE_ONE;
        }
        if (tag == Enumeration.VALUE_TWO.tag) {
            return Enumeration.VALUE_TWO;
        }
        throw new Error("Unexpected tag value");
    }

    /**
     * Write Enumeration to output stream
     */
    async writeTo(stream) {
        await stream.writeInt(this.tag);
    }

    [Symbol.for('nodejs.util.inspect.custom')]() {
        return this.name;
    }
}

module.exports = Enumeration;